tmt/steps/discover/fmf.py

```diff
@@ -561,11 +561,16 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
                 Command(
                     'git', 'log', '--format=', '--stat', '--name-only', f"{modified_ref}..HEAD"
                     ), cwd=self.testdir)
+            modified = None
             if output.stdout:
                 directories = [os.path.dirname(name) for name in output.stdout.split('\n')]
                 modified = {f"^/{re.escape(name)}" for name in directories if name}
                 self.debug(f"Limit to modified test dirs: {modified}", level=3)
                 names.extend(modified)
+            if modified is None:
+                # Nothing was modified, do not select anything
+                self._tests: list['tmt.Test'] = []
+                return
```

Comment:
I don't believe "leaking" `modified` outside of that condition is needed. IIUIC, it's scope is just that `if output.stdout` block, and nothing else. So the test can be performed there as well, as there is nothing outside of that block that can affect the `modified` value. How about this then, would it achieve the same outcome? In your code it's added to `names`, but if it was empty, `names` would not be used anyway, so...

```python
if output.stdout:
    directories = [os.path.dirname(name) for name in output.stdout.split('\n')]
    modified = {f"^/{re.escape(name)}" for name in directories if name}

    if not modified:
        # Nothing was modified, do not select anything
        self._tests: list['tmt.Test'] = []
        return

    self.debug(f"Limit to modified test dirs: {modified}", level=3)
    names.extend(modified)
```

Reply:
Good point. Fixed in fbfd0e07.
btw, any idea why `self._tests: list['tmt.Test'] = []` is necessary? mypy could accept it without such line (`error: Need type annotation for "_tests" (hint: "_tests: List[<type>] = ...")  [var-annotated]`
but isn't bothered with similar construct in `if self.is_dry_run` few lines below it missing the type.